platform/engineering/design-docs/2021-01-13-web-component-library.md

```diff
@@ -0,0 +1,337 @@
+
+### Testing Plan
+
+Each Component will have:
+
+- E2E tests written using Jest ([provided by Stencil](https://stenciljs.com/docs/testing-overview))
+- One or more stories in Storybook to validate appearance/behavior
+
+Unit tests will be written for utility/library functions, and for components where appropriate.
+
+As of mid-January 2021, Stencil provides an experimental screenshot visual diff testing functionality. If this becomes stable, we will consider adopting it.
+
+### Logging
+
+N/A
+
+### Debugging
+
+Storybook will be a useful tool for debugging and development. Stencil also comes with its own development server which can be used.
+
+There will be no special provisions for debugging provided for consumers of the WC library. Right now there is no plan to implement Sentry logging in the components.
+
+### Caveats
+
+To be determined.
+
+### Security Concerns
+
+N/A
+
+### Privacy Concerns
+
+N/A
+
+### Open Questions and Risks
+
+How will we ensure that each Component meets accessibility standards?
+
+How will we measure performance of Web Components, lazy-loading, etc?
+
+#### Risks
+
+##### Tech debt
+
+Even though Web Components are already a feature provided by the browser, this is a feature that the VA Design System hasn't yet taken advantage of. By introducing this "new" technology, there is a possibility of this just turning into technical debt if it exists alongside the current `component-library`, `formation` component styles, and Liquid templates for components. In order to avoid this, we want to be agressive about this migration and get out of the transitionary period as quickly and feasibly as we can.
+
+A Web Component library will be most effective and easiest to maintain if it is the _only_ component library.
+
+##### Deprecation
+
+An additional risk would be if certain features of the Web Component standard are deprecated or otherwise become obselete. We are already in a similar situation with React components - our `component-library` consists of many "legacy" class-based components rather than functional components which have the option to use hooks.
+
+##### Abandoned development
+
+Web Components are a feature that is already built into modern browsers, but they don't have as wide of an industry adoption as a framework like React. Despite this, it is unlikely that the Web Component standard will be completely abandoned. There are many large companies which are already developing products using Web Components:
+
+- Youtube
+- GitHub
+- Xfinity (Comcast)
+- [Apple Music](https://hackernoon.com/apple-cements-the-unlikely-rise-of-web-components-6b1d3g1t)
+- [Adobe](https://opensource.adobe.com/spectrum-web-components/)
+
+There aren't any published articles detailing Web Component adoption by the companies without links, but Web Components can be found on their websites.
```

Comment:
This is a new section added in response to your comment:

> While I, like many other open source and gov people, like the idea of platform Web Components, I am not clear that I see industry adoption of this approach. I worry that down the line we'll need to pivot again because this will be a dead-end. What would reassure me is finding out that some major organization are using it.